### PR TITLE
Add `tribe_events_month_has_events_filtered` function and filter

### PR DESCRIPTION
**Pull request by juanfra**
🎫 https://central.tri.be/issues/67607

Not happy with the results. Been trying many different things, and this one seemed to be to most cost-efficient one. Not sure if you had something else in mind, @barryhughes?

### CODE REVIEW

File: src/Tribe/Template/Month.php [hunk 1]
```diff
@@ -1234,5 +1234,50 @@ public function has_events() {
...
+		/**
+		 * Check if the month has events when all the filters have been applied.
+		 *
+		 * @since TBD
+		 *
+		 * @return bool
+		 */
+		public function has_events_filtered() {
+
+			if ( 'month' === Tribe__Events__Main::instance()->displaying ) {
```
Comment (bordoni) [inline src/Tribe/Template/Month.php]: Lets invert this logic, to `return false;` if `!==` than `month`, avoids indentation.

File: src/Tribe/Template/Month.php [hunk 2]
```diff
@@ -1234,5 +1234,50 @@ public function has_events() {
...
+
+				if ( ! $wp_query = tribe_get_global_query_object() ) {
+					return;
```
Comment (bordoni) [inline src/Tribe/Template/Month.php]: let's return `false` here.

File: src/functions/template-tags/month.php [hunk 3]
```diff
@@ -195,6 +196,17 @@ function tribe_events_month_has_events() {
 		return apply_filters( 'tribe_events_month_has_events', false );
 	}
 
+	/**
+	 * Returns whether there are any events in the month,
+	 * with the filtered results.
+	 *
```
Comment (bordoni) [inline src/functions/template-tags/month.php]: Add a `@since`

File: src/functions/template-tags/month.php [hunk 4]
```diff
@@ -195,6 +196,17 @@ function tribe_events_month_has_events() {
...
+	 * @return bool
+	 * @see Tribe__Events__Template__Month::has_events_filtered()
+	 **/
+	function tribe_events_month_has_events_filtered() {
+		return apply_filters( 'tribe_events_month_has_events_filtered', Tribe__Events__Template__Month::has_events_filtered() );
```
Comment (bordoni) [inline src/functions/template-tags/month.php]: Lets add a docblock.

File: src/functions/template-tags/month.php (hunk 3 above)
Comment (juanfra) [reply src/functions/template-tags/month.php]: This function was already there. I will add the since of the version where it was introduced.

**Pull request merged**